src/plugins/chart_expressions/expression_xy/public/expression_renderers/xy_chart_renderer.tsx

```diff
@@ -76,11 +86,60 @@ const extractCounterEvents = (originatingApp: string, layers: CommonXYLayerConfi
       }
     );
 
+    // Multiple axes configured on the same side of the chart
+    const multiAxisSide = Object.values(
+      (yAxisConfigs ?? []).reduce<Record<string, number>>((acc, item) => {
+        if (item.position) {
+          acc[item.position] = (acc[item.position] ?? 0) + 1;
+        }
+        return acc;
+      }, {})
+    ).find((i) => i > 1);
+
+    // There are multiple "split series" aggs in an xy chart and they are not all terms but other aggs
+    const multiSplitNonTerms = new Set(
+      (dataLayer.splitAccessors ?? [])
+        .map(
+          (splitAccessor) =>
+            getColumnByAccessor(splitAccessor, dataLayer.table.columns)?.meta?.params?.id
```

Comment:
how is this checking whether it's terms or not?

I probably worded that badly, but the thing we are checking for is whether these two conditions hold true together:
* There is more than one split series
* Not all of them a terms aggs

If there are multiple terms splits, it shouldn't trigger
If there is one split which is using filters or something else, it shouldn't trigger
If there is a terms split and a filters split, it should trigger

Reply:
@flash1293 is the following statement correct:

> If there are multiple splits:` Range + Filters`,  it should trigger

If yes, current code fully match your requirements and we don't need to chec whether it's `terms` or not